static/app/views/alerts/rules/issue/ticketRuleModal.spec.tsx

```diff
@@ -214,6 +214,19 @@ describe('ProjectAlerts -> TicketRuleModal', function () {
       await submitSuccess();
     });
 
+    it('should persist non-choice value when the modal is reopened', async function () {
+      const textField: IssueConfigField = {
+        label: 'Text Field',
+        required: true,
+        type: 'string',
+        name: 'textField',
+      };
+      renderComponent({data: {textField: 'foo'}}, textField);
+
+      expect(screen.getByRole('textbox', {name: 'Text Field'})).toHaveValue('foo');
+      await submitSuccess();
+    });
+
```

Comment:
just curious, what part of this is mocking the modal being reopened? i only see the component being rendered once

Reply:
It's a bit weird, but the presence of `data` values are mimicking what would occur when values are previously set and the dialog is reopened.